src/v2/System/useFeatureFlag.tsx

```diff
@@ -1,18 +1,39 @@
 import { useSystemContext } from "v2/System"
+import { Variant } from "unleash-client"
 
-export function useFeatureFlag(flagName: string): boolean | null {
+export type FeatureFlags = Record<string, FeatureFlagDetails>
+
+interface FeatureFlagDetails {
+  flagEnabled: boolean
+  variant: Variant
+}
+
+export function useFeatureFlag(featureName: string): boolean | null {
   const { featureFlags } = useSystemContext()
-  if (featureFlags === undefined) {
-    console.error("[Force] Error: featureFlags is undefined in SystemContext")
+  const flagEnabled = featureFlags?.[featureName]?.flagEnabled
+
+  if (flagEnabled === undefined) {
```

Comment:
`!flagEnabled` will check for `false, undefined, null, 0`

Reply:
Correct. And we want to return out false, when `featureFlags?.[featureName]?.flagEnabled = false`.

Reply:
Cool 👍  to be clear, just meaning if we were to do `flagEnabled === undefined` we might miss some case, and `!thing` is a quick way to match a variety of falsy cases.

Reply:
For sure! My thinking was the safe navigational operator will catch and return out `undefined` for anything nullish, allowing us to rely on `undefined` for the subset of falsy cases we're concerned with catching.